hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -504,3 +504,98 @@ def test_minimize_negative_tuple_axes(ndim, data):
         nps.valid_tuple_axes(ndim, min_size, max_size), lambda x: all(i < 0 for i in x)
     )
     assert len(smallest) == min_size
+
+
+@given(shape=nps.array_shapes(min_side=0, min_dims=0, max_dims=3), data=st.data())
+def test_broadcastable_bounds_are_satisfied(shape, data):
+    min_dim = data.draw(st.integers(0, 5), label="min_dim")
+    max_dim = data.draw(st.integers(min_dim, 5), label="max_dim")
+    min_side = data.draw(st.integers(0, 10), label="min_side")
+    max_side = data.draw(st.integers(min_side, 10), label="max_side")
+    bshape = data.draw(
+        nps.broadcastable_shapes(
+            shape,
+            min_side=min_side,
+            max_side=max_side,
+            min_dims=min_dim,
+            max_dims=max_dim,
+        ),
+        label="broadcastable shape",
+    )
+
+    n_leading = max((len(bshape) - len(shape)), 0)
+    assert isinstance(bshape, tuple) and all(isinstance(s, int) for s in bshape)
+    assert min_dim <= len(bshape) <= max_dim
+    assert all(min_side <= s <= max_side for s in bshape[:n_leading])
+
+
+@given(
+    shape=nps.array_shapes(min_dims=0, max_dims=3, min_side=0, max_side=10),
+    data=st.data(),
+)
+def test_broadcastable_shape_can_broadcast(shape, data):
+    min_dim = data.draw(st.integers(0, 5), label="min_dim")
+    max_dim = data.draw(st.integers(min_dim, 5), label="max_dim")
+    min_side = data.draw(st.integers(0, 10), label="min_side")
+    max_side = data.draw(st.integers(min_side, 10), label="max_side")
+    broadcastable_shape = data.draw(
+        nps.broadcastable_shapes(
+            shape,
+            min_side=min_side,
+            max_side=max_side,
+            min_dims=min_dim,
+            max_dims=max_dim,
+        ),
+        label="broadcastable_shapes",
+    )
+    a = np.empty(shape)
```

Comment:
`np.empty` makes me nervous, even when it's "obviously safe".  For the sake of peace-of-mind-when-grepping, maybe `np.zeros` here instead?